x-pack/plugins/security_solution/public/onboarding/components/onboarding_body/cards/integrations/package_list_grid.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Suspense, useCallback, useEffect, useRef, useState } from 'react';
+
+import { EuiButtonGroup, EuiFlexGroup, EuiFlexItem, EuiSkeletonText } from '@elastic/eui';
+import type { AvailablePackagesHookType, IntegrationCardItem } from '@kbn/fleet-plugin/public';
+import { noop } from 'lodash';
+
+import { css } from '@emotion/react';
+import { PackageList } from './utils';
+import { useIntegrationCardList, useTabMetaData } from './hooks';
+import {
+  useStoredIntegrationSearchTerm,
+  useStoredIntegrationTabId,
+} from '../../../../hooks/use_stored_state';
+import { useOnboardingContext } from '../../../onboarding_context';
+import {
+  DEFAULT_TAB,
+  INTEGRATION_TABS,
+  LOADING_SKELETON_HEIGHT,
+  SCROLL_ELEMENT_ID,
+  SEARCH_FILTER_CATEGORIES,
+  WITHOUT_SEARCH_BOX_HEIGHT,
+  WITH_SEARCH_BOX_HEIGHT,
+} from './const';
+
+interface WrapperProps {
+  useAvailablePackages: AvailablePackagesHookType;
+}
+
+const emptyStateStyles = { paddingTop: '16px' };
+export const PackageListGrid = React.memo(({ useAvailablePackages }: WrapperProps) => {
+  const { spaceId } = useOnboardingContext();
+  const scrollElement = useRef<HTMLDivElement>(null);
+  const [selectedTabId, setSelectedTabIdToStorage] = useStoredIntegrationTabId(
+    spaceId,
+    DEFAULT_TAB.id
+  );
+  const [searchTermFromStorage, setSearchTermToStorage] = useStoredIntegrationSearchTerm(spaceId);
+  const [toggleIdSelected, setToggleIdSelected] = useState<string>(selectedTabId);
```

Comment:
Why do we need 2 states for the `selectedTabId`?

Reply:
The first one should come from the store, I'll rename it to selectedTabIdFromStorage to make it clearer.

Reply:
Yes. But, as far as I know, the only limitation with the `useLocalStorage` state is that setting the value with a function (to get the current value) like: `setSelectedTabId((currentValue) => // next state )` does not work well. I don't see we need to do this with it. So I think we are okay using only:
```
const [selectedTabId, setSelectedTabId] = useStoredIntegrationTabId(
    spaceId,
    DEFAULT_TAB.id
  );
```
And removing the `const [toggleIdSelected, setToggleIdSelected] = useState<string>(selectedTabId);`

Or is there any other reason that I am not aware of?